### PR TITLE
WIP: Sidecar and pyinstaller

### PR DESCRIPTION
**Pull request by BudLight-Year**
Made new branch to avoid merge conflicts. Copy and pasting previous description below.

Using pyinstaller to create executable for python code so we no longer need a venv and pyo3.

Currently works for windows, needs mac testing before pull.

**Instructions**
To create the executable navigate into src-tauri/src. 
Make sure you have pyinstaller installed. (pip3 install pyinstaller)

If you are using windows you can use the spec file for the audio_analyzer python file in there.

If you are on mac you will have to generate a spec file using the following command: 
pyinstaller --onefile audio_analyzer.py
After the spec file is generated open it and ensure librosa and numpy are in the hiddenimports. 
It should look like this:
hiddenimports=['librosa',  'numpy'],

Then run the following command:
pyinstaller audio_analyzer.spec

This will create your operating systems version of the executable in dist inside src-tauri/src
Create a dist folder inside src-tauri and place the executable inside there.

Run: cargo build

The output will generate an error informing you what to rename your executable to.
On windows it looks like:
audio_analyzer-x86_64-pc-windows-msvc.exe

Then cargo build should work and you can run the app to test functionality.

### CODE REVIEW

**Comment by daskimmel** [comment]
Additional comments:

get_executable_path() will need 
- project_root.join("src-tauri").join("dist").join("audio_analyzer-aarch64-apple-darwin")

with your path to py executable defined in the last join

I think we can remove the 
```
      "resources": [
        "../python/windows"
      ]
```
from tauri.conf.json

at least on mac, this WILL NOT WORK with python version 3.12.0. updating to 3.12.1 made it work for me

**Comment by daskimmel** [comment]
Final comment, it would be nice to pass in a list of file_paths to the python file so that we only need to load up the executable once as that seems to be the slowest part

we also will need to make sure this runs with `pnpm run build`

File: src-tauri/tauri.conf.json [hunk 1]
```diff
@@ -54,6 +61,12 @@
         "icons/128x128@2x.png",
         "icons/icon.icns",
         "icons/icon.ico"
+      ],
+      "externalBin": [
+        "dist/audio_analyzer"
+      ],
+      "resources": [
+        "../python/windows"
       ]
```
Comment (daskimmel) [inline src-tauri/tauri.conf.json]: suggest to remove this, breaks the build on mac and works fine without it

**Comment by BudLight-Year** [comment]
closed in favor of #43

**Pull request closed without merging**